### ISSUE TITLE
S2I build failed at waiting for container

### ISSUE THREAD

**Issue by f-w** [BCDevOps/issues-and-solutions#5]
S2I build fails over 50% of chance. Error happens when waiting for a container. 

```
I0314 14:31:26.734519       1 sti.go:173] The value of ALLOWED_UIDS is [1-]
I0314 14:31:26.853863       1 docker.go:224] Image abbr/node@sha256:b220d202f010311904197c03c5bbc32b11a9a78d25c6657727b417576a83e9be available locally
I0314 14:31:26.867618       1 sti.go:195] Creating a new S2I builder with build config: "Builder Image:\t\t\tabbr/node@sha256:b220d202f010311904197c03c5bbc32b11a9a78d25c6657727b417576a83e9be\nSource:\t\t\t\tfile:///tmp/s2i-build617694700/upload/src#aa8d14feac6e38f908fa266fd485e26aaa1d4699\nOutput Image Tag:\t\t172.16.156.227:5000/bcdev/nodejs-mongodb-binary-src:latest\nEnvironment:\t\t\tOPENSHIFT_BUILD_NAME=nodejs-mongodb-binary-src-24,OPENSHIFT_BUILD_NAMESPACE=bcdev\nIncremental Build:\t\t\x1b[1menabled\x1b[0m\nIncremental Image Pull User:\tserviceaccount\nRemove Old Build:\t\tdisabled\nBuilder Pull Policy:\t\tif-not-present\nQuiet:\t\t\t\tdisabled\nLayered Build:\t\t\tdisabled\nWorkdir:\t\t\t/tmp/s2i-build617694700\nDocker NetworkMode:\t\tcontainer:f63ff72682200fde070f33286d2d6ec93cff7b96efa60e5f9ce3f4fff2f225e4\nDocker Endpoint:\t\tunix:///var/run/docker.sock\n"
I0314 14:31:26.883069       1 docker.go:224] Image abbr/node@sha256:b220d202f010311904197c03c5bbc32b11a9a78d25c6657727b417576a83e9be available locally
I0314 14:31:26.919703       1 sti.go:140] Preparing to build 172.16.156.227:5000/bcdev/nodejs-mongodb-binary-src:latest
I0314 14:31:27.024595       1 source.go:167] Receiving source from STDIN as archive
I0314 14:31:28.195922       1 docker.go:224] Image abbr/node@sha256:b220d202f010311904197c03c5bbc32b11a9a78d25c6657727b417576a83e9be available locally
I0314 14:31:28.196056       1 docker.go:344] Image contains io.openshift.s2i.scripts-url set to 'image:///home/node/s2i/bin'
I0314 14:31:28.196119       1 download.go:57] Using image internal scripts from: image:///home/node/s2i/bin/assemble
I0314 14:31:28.196136       1 download.go:57] Using image internal scripts from: image:///home/node/s2i/bin/run
I0314 14:31:28.209804       1 docker.go:224] Image abbr/node@sha256:b220d202f010311904197c03c5bbc32b11a9a78d25c6657727b417576a83e9be available locally
I0314 14:31:28.209826       1 docker.go:344] Image contains io.openshift.s2i.scripts-url set to 'image:///home/node/s2i/bin'
I0314 14:31:28.209850       1 download.go:57] Using image internal scripts from: image:///home/node/s2i/bin/save-artifacts
I0314 14:31:28.209913       1 sti.go:221] Using assemble from upload/src/.sti/bin
I0314 14:31:28.209926       1 sti.go:221] Using run from upload/src/.sti/bin
I0314 14:31:28.209934       1 sti.go:221] Using save-artifacts from upload/src/.sti/bin
I0314 14:31:28.216895       1 docker.go:224] Image 172.16.156.227:5000/bcdev/nodejs-mongodb-binary-src:latest available locally
I0314 14:31:28.219648       1 sti.go:146] Existing image for tag 172.16.156.227:5000/bcdev/nodejs-mongodb-binary-src:latest detected for incremental build
I0314 14:31:28.219666       1 sti.go:151] Performing source build from file:///tmp/s2i-build617694700/upload/src#aa8d14feac6e38f908fa266fd485e26aaa1d4699
I0314 14:31:28.219786       1 sti.go:370] Saving build artifacts from image 172.16.156.227:5000/bcdev/nodejs-mongodb-binary-src:latest to path /tmp/s2i-build617694700/upload/artifacts
I0314 14:31:28.225171       1 docker.go:390] Both scripts and untarred source will be placed in '/tmp'
I0314 14:31:28.225195       1 docker.go:529] Creating container using config: {Hostname: Domainname: User:1001 Memory:0 MemorySwap:0 CPUShares:0 CPUSet: AttachStdin:false AttachStdout:true AttachStderr:false PortSpecs:[] ExposedPorts:map[] Tty:false OpenStdin:false StdinOnce:false Env:[] Cmd:[/tmp/scripts/save-artifacts] DNS:[] Image:172.16.156.227:5000/bcdev/nodejs-mongodb-binary-src:latest Volumes:map[] VolumeDriver: VolumesFrom: WorkingDir: MacAddress: Entrypoint:[] NetworkDisabled:false SecurityOpts:[] OnBuild:[] Mounts:[] Labels:map[]}
I0314 14:31:29.253119       1 docker.go:543] Attaching to container
I0314 14:31:29.254607       1 docker.go:549] Starting container
I0314 14:35:02.187608       1 tar.go:250] Done extracting tar stream
I0314 14:35:02.187711       1 docker.go:479] Waiting for container
E0314 14:35:02.217245       1 docker.go:471] Unable to attach container with {77425c09f0b6579466cbc0e6723a42957b375c2589d47e9eabced327b7f82a96 <nil> 0xc2083c2128 0xc2083c2138 false true false true true 0xc208083740 false}
I0314 14:35:02.351110       1 docker.go:481] Container wait returns with 0 and <nil>
I0314 14:35:02.351161       1 docker.go:488] Container exited
I0314 14:35:02.636057       1 sti.go:164] Running "assemble" in "172.16.156.227:5000/bcdev/nodejs-mongodb-binary-src:latest"
I0314 14:35:02.636114       1 sti.go:412] Using image name abbr/node@sha256:b220d202f010311904197c03c5bbc32b11a9a78d25c6657727b417576a83e9be
I0314 14:35:02.636197       1 sti.go:416] No .sti/environment provided (no environment file found in application sources)
I0314 14:35:02.733721       1 docker.go:390] Both scripts and untarred source will be placed in '/tmp'
I0314 14:35:02.733811       1 docker.go:529] Creating container using config: {Hostname: Domainname: User: Memory:0 MemorySwap:0 CPUShares:0 CPUSet: AttachStdin:false AttachStdout:true AttachStderr:false PortSpecs:[] ExposedPorts:map[] Tty:false OpenStdin:true StdinOnce:true Env:[OPENSHIFT_BUILD_NAME=nodejs-mongodb-binary-src-24 OPENSHIFT_BUILD_NAMESPACE=bcdev] Cmd:[/bin/sh -c tar -C /tmp -xf - && /tmp/scripts/assemble] DNS:[] Image:abbr/node@sha256:b220d202f010311904197c03c5bbc32b11a9a78d25c6657727b417576a83e9be Volumes:map[] VolumeDriver: VolumesFrom: WorkingDir: MacAddress: Entrypoint:[] NetworkDisabled:false SecurityOpts:[] OnBuild:[] Mounts:[] Labels:map[]}
I0314 14:35:04.958784       1 docker.go:543] Attaching to container
I0314 14:35:05.009076       1 docker.go:549] Starting container
I0314 14:35:06.081737       1 docker.go:479] Waiting for container
unexpected EOFerror: The build bcdev/nodejs-mongodb-binary-src-24 status is "Failed"
Build step 'Execute shell' marked build as failure
Finished: FAILURE
```

**Comment by f-w** [comment]
```
status:
  phase: Pending
  message: "Query container info failed with error (API error (500): Unknown device 21e30ea2a8689df38de8b3f4a21f08c0c116be93deac1506c36b1fe6c7facf3c\n)"
  reason: GeneralError
  startTime: '2016-03-17T16:48:12Z'
```

**Comment by ckayfish** [comment]
Maybe related to this: https://github.com/kubernetes/kubernetes/issues/14404? A kubernetes update on march 6th addresses the kubelet trying to build in a failed container 
https://github.com/kubernetes/kubernetes/pull/22576

**Comment by sferich888** [comment]
This looks similar to https://bugzilla.redhat.com/show_bug.cgi?id=1331237, which found a race condition in docker. I found this gitrepo, when searching for workarounds to the issue, so sorry if this is a bit out of context but I figured I would try and help others out, as it also looks like your still searching for a solution? https://github.com/projectatomic/docker/pull/171/files is the fix, for the BZ linked above.
